rust-runtime/aws-smithy-types/src/lib.rs

```diff
@@ -12,6 +12,7 @@
     rust_2018_idioms,
     unreachable_pub
 )]
+#![allow(clippy::derive_partial_eq_without_eq)] // TODO: derive Eq for appropriate types
```

Comment:
Let's just add `Eq` to the derives for `Blob`

Reply:
#2223 has added `Eq`. So I just allow the lint.